cordova-plugin-one/src/android/OnePlugin.java

```diff
@@ -8,419 +8,405 @@
-            one.sendProperties(interactionPath, propertiesMap);
+        HashMap<String, String> propertiesMap = getPropertiesFromJSONObject(jsonObject);
+        if (propertiesMap != null && propertiesMap.size() > 0) {
+          sendProperties(args.getString(0), propertiesMap);
+          callbackContext.success();
+        } else {
+          callbackContext.error("No properties to send");
         }
-    }
-    
-    public void sendBaseTouchpointProperties(HashMap<String, String> propertiesMap) {
-        if (one != null && propertiesMap != null) {
-            Log.d(LOG_TAG, "Sending base Touchpoint properties: " + propertiesMap);
-            one.sendBaseTouchpointProperties(propertiesMap);
+      } catch (Exception e) {
+        Log.e(LOG_TAG, "Failed to send properties.", e);
+        callbackContext.error("Failed to send properties: " + e.getLocalizedMessage());
+      }
+      return true;
+    } else if ("sendBaseTouchpointProperties".equals(action)) {
+      try {
+        JSONObject jsonObject = args.getJSONObject(0);
+        HashMap<String, String> propertiesMap = getPropertiesFromJSONObject(jsonObject);
+        if (propertiesMap != null && propertiesMap.size() > 0) {
+          sendBaseTouchpointProperties(propertiesMap);
+          callbackContext.success();
+        } else {
+          callbackContext.error("No base Touchpoint properties to send");
         }
+      } catch (Exception e) {
+        Log.e(LOG_TAG, "Failed to send base Touchpoint properties.", e);
+        callbackContext.error("Failed to send base Touchpoint properties: " + e.getLocalizedMessage());
+      }
+      return true;
+    } else if ("sendResponseCode".equals(action)) {
+      try {
+        OneResponseCodeRequest request = new OneResponseCodeRequest.Builder()
+          .interactionPath(new OneInteractionPath(URI.create(args.optString(1))))
+          .responseCode(new OneResponseCode(args.getString(0)))
+          .build();
+        One.sendResponseCode(request).enqueue(null);
+      } catch (Exception e) {
+        Log.e(LOG_TAG, "Failed to send response code.", e);
+        callbackContext.error("Failed to send response code: " + e.getLocalizedMessage());
+      }
+      return true;
+    } else if ("getTid".equals(action)) {
+      callbackContext.success(getTid());
+      return true;
+    } else if ("clearUserProfile".equals(action)) {
+      clearUserProfile();
+      if (getTid() == null || getTid().length() == 0) {
+        callbackContext.success();
+      } else {
+        callbackContext.error("Failed to clear user profile");
+      }
+      return true;
+    } else if ("sendInteractionForOutboundLink".equals(action)) {
+      try {
+        String stringURL = args.optString(0);
+        URL url = new URL(stringURL);
+        sendInteractionForOutboundLink(url);
+      } catch (Exception e) {
+        Log.e(LOG_TAG, "Failed to send interaction for outbound link.", e);
```

Comment:
Some exception use the log tag, some don't. Do we need both? 
1. If so, why? 
2. Why don't add the same setup everywhere?
3. Pull the message in a string and reuse to not duplicate.

Reply:
Missed one other spot that was still using printStackTrace.  Updated all exceptions to Log error with LOG_TAG for consistency.